### PR TITLE
Handle location descriptors in createLocation

### PR DESCRIPTION
**Pull request by taion**

### CODE REVIEW

File: modules/__tests__/describeQueries.js [hunk 1]
```diff
@@ -201,6 +201,37 @@ function describeQueries(createHistory) {
...
+      describe('in createLocation', function () {
+        it('works with string', function () {
+          const location = history.createLocation('/the/path?the=query')
+
+          expect(location.pathname).toEqual('/the/path')
+          expect(location.query).toEqual({ the: 'query' })
+          expect(location.search).toEqual('?the=query')
+        })
+
+        it('works with object with query', function () {
```
Comment (taion) [inline modules/__tests__/describeQueries.js]: This case will not work without this patch.

**Comment by taion** [comment]
@mjackson Can you look at this when you get a chance, and maybe cut 2.0.0-rc2 off this? It's blocking a few more API cleanups we need for the router.

**Comment by taion** [comment]
On second thought, this might be the wrong implementation; instead, we're probably better off just passing fields through `createLocation`.

**Comment by taion** [comment]
Nope, never mind, that doesn't work.

**Comment by taion** [comment]
Updated for better forward-compat with https://github.com/rackt/react-router/issues/2859.

**Comment by timdorr** [comment]
Looks like the changes merge conflict. That's why I tend to leave docs out of PRs. For now, just rebase and get it green :smile:

**Comment by taion** [comment]
Eh, just rebased for now.

**Pull request merged**